[BUG] test_case_when failed on Databricks 7.3 nightly build

```
[2021-08-25T19:37:41.778Z] =================================== FAILURES ===================================
[2021-08-25T19:37:41.778Z] [31m[1m______ test_case_when[Map(String(not_null),Map(String(not_null),String))] ______[0m
[2021-08-25T19:37:41.778Z] [gw1] linux -- Python 3.7.10 /databricks/conda/envs/databricks-ml-gpu/bin/python
[2021-08-25T19:37:41.778Z] 
[2021-08-25T19:37:41.778Z] data_gen = Map(String(not_null),Map(String(not_null),String))
[2021-08-25T19:37:41.778Z] 
[2021-08-25T19:37:41.778Z]     [37m@pytest[39;49;00m.mark.parametrize([33m'[39;49;00m[33mdata_gen[39;49;00m[33m'[39;49;00m, all_gens + all_nested_gens, ids=idfn)
[2021-08-25T19:37:41.778Z]     [94mdef[39;49;00m [92mtest_case_when[39;49;00m(data_gen):
[2021-08-25T19:37:41.778Z]         num_cmps = [94m20[39;49;00m
[2021-08-25T19:37:41.778Z] >       s1 = gen_scalar(data_gen, force_no_nulls=[95mnot[39;49;00m [96misinstance[39;49;00m(data_gen, NullGen))
[2021-08-25T19:37:41.778Z] 
[2021-08-25T19:37:41.779Z] [1m[31m../../src/main/python/conditionals_test.py[0m:72: 
[2021-08-25T19:37:41.779Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2021-08-25T19:37:41.779Z] [1m[31m../../src/main/python/data_gen.py[0m:696: in gen_scalar
[2021-08-25T19:37:41.779Z]     v = [96mlist[39;49;00m(gen_scalars(data_gen, [94m1[39;49;00m, seed=seed, force_no_nulls=force_no_nulls))
[2021-08-25T19:37:41.779Z] [1m[31m../../src/main/python/data_gen.py[0m:692: in <genexpr>
[2021-08-25T19:37:41.779Z]     [94mreturn[39;49;00m (_mark_as_lit(src.gen(force_no_nulls=force_no_nulls), data_type) [94mfor[39;49;00m i [95min[39;49;00m [96mrange[39;49;00m([94m0[39;49;00m, count))
[2021-08-25T19:37:41.779Z] [1m[31m../../src/main/python/data_gen.py[0m:666: in _mark_as_lit
[2021-08-25T19:37:41.779Z]     col_array.append(_mark_as_lit(data[k], data_type.valueType))
[2021-08-25T19:37:41.779Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2021-08-25T19:37:41.779Z] 
[2021-08-25T19:37:41.779Z] data = None, data_type = MapType(StringType,StringType,true)
[2021-08-25T19:37:41.779Z] 
[2021-08-25T19:37:41.779Z]     [94mdef[39;49;00m [92m_mark_as_lit[39;49;00m(data, data_type):
[2021-08-25T19:37:41.779Z]         [90m# To support nested types, 'data_type' is required.[39;49;00m
[2021-08-25T19:37:41.779Z]         [94massert[39;49;00m data_type [95mis[39;49;00m [95mnot[39;49;00m [94mNone[39;49;00m
[2021-08-25T19:37:41.779Z]     
[2021-08-25T19:37:41.779Z]         [94mif[39;49;00m [96misinstance[39;49;00m(data_type, ArrayType):
[2021-08-25T19:37:41.779Z]             [94massert[39;49;00m [96misinstance[39;49;00m(data, [96mlist[39;49;00m)
[2021-08-25T19:37:41.779Z]             [90m# Sadly you cannot create a literal from just an array in pyspark[39;49;00m
[2021-08-25T19:37:41.779Z]             [94mreturn[39;49;00m f.array([_mark_as_lit(x, data_type.elementType) [94mfor[39;49;00m x [95min[39;49;00m data])
[2021-08-25T19:37:41.779Z]         [94melif[39;49;00m [96misinstance[39;49;00m(data_type, StructType):
[2021-08-25T19:37:41.779Z]             [94massert[39;49;00m [96misinstance[39;49;00m(data, [96mtuple[39;49;00m) [95mand[39;49;00m [96mlen[39;49;00m(data) == [96mlen[39;49;00m(data_type.fields)
[2021-08-25T19:37:41.779Z]             [90m# Sadly you cannot create a literal from just a dict/tuple in pyspark[39;49;00m
[2021-08-25T19:37:41.779Z]             children = [96mzip[39;49;00m(data, data_type.fields)
[2021-08-25T19:37:41.779Z]             [94mreturn[39;49;00m f.struct([_mark_as_lit(x, fd.dataType).alias(fd.name) [94mfor[39;49;00m x, fd [95min[39;49;00m children])
[2021-08-25T19:37:41.779Z]         [94melif[39;49;00m [96misinstance[39;49;00m(data_type, DateType) [95mand[39;49;00m data [95mis[39;49;00m [95mnot[39;49;00m [94mNone[39;49;00m:
[2021-08-25T19:37:41.779Z]             [90m# Due to https://bugs.python.org/issue13305 we need to zero pad for years prior to 1000,[39;49;00m
[2021-08-25T19:37:41.779Z]             [90m# but this works for all of them[39;49;00m
[2021-08-25T19:37:41.779Z]             dateString = data.strftime([33m"[39;49;00m[33m%[39;49;00m[33mY-[39;49;00m[33m%[39;49;00m[33mm-[39;49;00m[33m%d[39;49;00m[33m"[39;49;00m).zfill([94m10[39;49;00m)
[2021-08-25T19:37:41.779Z]             [94mreturn[39;49;00m f.lit(dateString).cast(data_type)
[2021-08-25T19:37:41.779Z]         [94melif[39;49;00m [96misinstance[39;49;00m(data_type, MapType):
[2021-08-25T19:37:41.779Z] >           [94massert[39;49;00m [96misinstance[39;49;00m(data, [96mdict[39;49;00m)
[2021-08-25T19:37:41.779Z] [1m[31mE           AssertionError[0m
[2021-08-25T19:37:41.779Z] 
[2021-08-25T19:37:41.779Z] [1m[31m../../src/main/python/data_gen.py[0m:661: AssertionError
```

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -643,6 +643,9 @@ def _mark_as_lit(data, data_type):
     # To support nested types, 'data_type' is required.
     assert data_type is not None
 
+    if data is None:
+        return f.lit(data).cast(data_type)
+
     if isinstance(data_type, ArrayType):
         assert isinstance(data, list)
         # Sadly you cannot create a literal from just an array in pyspark
@@ -652,7 +655,7 @@ def _mark_as_lit(data, data_type):
         # Sadly you cannot create a literal from just a dict/tuple in pyspark
         children = zip(data, data_type.fields)
         return f.struct([_mark_as_lit(x, fd.dataType).alias(fd.name) for x, fd in children])
-    elif isinstance(data_type, DateType) and data is not None:
+    elif isinstance(data_type, DateType):
         # Due to https://bugs.python.org/issue13305 we need to zero pad for years prior to 1000,
         # but this works for all of them
         dateString = data.strftime("%Y-%m-%d").zfill(10)

